backend/terabox/terabox.go

```diff
@@ -0,0 +1,753 @@
+					encoder.EncodeBackQuote |
+					encoder.EncodeDoubleQuote |
+					encoder.EncodeLtGt |
+					encoder.EncodeLeftSpace |
+					encoder.EncodeInvalidUtf8),
+			}},
+	})
+}
+
+// Options defines the configuration for this backend
+type Options struct {
+	// AccessToken  string               `config:"access_token"`
+	Cookie            string               `config:"cookie"`
+	DeletePermanently bool                 `config:"delete_permanently"`
+	UploadThreads     uint8                `config:"upload_threads"`
+	UserAgent         string               `config:"user_agent"`
+	DebugLevel        uint8                `config:"debug_level"`
+	Enc               encoder.MultiEncoder `config:"encoding"`
+}
+
+//
+//------------------------------------------------------------------------------------------------------------------------
+//
+
+// Fs is the interface a cloud storage system must provide
+type Fs struct {
+	root     string
+	name     string
+	opt      *Options
+	features *fs.Features
+	client   *rest.Client
+	// pacer    *fs.Pacer
+
+	origRoot     string
+	origRootItem *api.Item
+
+	baseURL     string
+	notFirstRun bool
+
+	// upload host should be got only once
+	uploadHost   string
+	uploadHostMX sync.Once
+
+	// official access [added for future releases]
+	accessToken string
+
+	// unofficial access [web token required for upload]
+	jsToken string
+
+	isPremium   bool
+	isPremiumMX sync.Once
+}
+
+// NewFs makes a new Fs object from the path
+//
+// The path is of the form remote:path
+//
+// Remotes are looked up in the config file.  If the remote isn't
+// found then NotFoundInConfigFile will be returned.
+//
+// On Windows avoid single character remote names as they can be mixed
+// up with drive letters.
+//
+// copyto: for srcFS the root is with the file name, for dstFS last segment (filename) will be cutted
+func NewFs(ctx context.Context, name string, root string, config configmap.Mapper) (fs.Fs, error) {
+	opt := new(Options)
+	if err := configstruct.Set(config, opt); err != nil {
+		return nil, err
+	}
+	opt.Cookie = valuedCookie(opt.Cookie)
+
+	debug(opt, 1, "NewFS %s; %s; %+v;", name, root, opt)
+
+	if len(root) > 0 {
+		if root[0:1] == "." {
+			root = root[1:]
+		}
+
+		if root[0:1] != "/" {
+			root = "/" + root
+		}
+
+	} else if root == "" {
+		root = "/"
+	}
+
+	f := &Fs{
+		name:     name,
+		root:     root,
+		origRoot: root, // save origin root, because it can change, if path is file
+		opt:      opt,
+		// pacer: fs.NewPacer(ctx, pacer.NewDefault(pacer.MinSleep(minSleep), pacer.MaxSleep(maxSleep), pacer.DecayConstant(decayConstant), pacer.AttackConstant(attackConstant))),
+
+		baseURL: baseURL,
+		// jsToken: "",
+	}
+
+	f.features = (&fs.Features{
+		ReadMetadata:            true,
+		CanHaveEmptyDirectories: true,
+	}).Fill(ctx, f)
+
+	newCtx, clientConfig := fs.AddConfig(ctx)
+	if opt.UserAgent != "" {
+		clientConfig.UserAgent = opt.UserAgent
+	}
+	clientConfig.Timeout = 5 * fs.Duration(time.Second)
```

Comment:
Is hardcoding the timeout necessary? This prevents users from customizing it with `--timeout`.
In fact, 5 seconds is too short for some low-quality networks.